pkgs/development/haskell-modules/hackage-packages.nix

```diff
@@ -122375,6 +122375,7 @@ self: {
          base containers directory extensible-exceptions filepath mtl
          process unix utf8-string X11
        ];
+       patches = [ ./xmonad-nix.patch ];
```

Comment:
The `patch` attribute comes from this code: https://github.com/NixOS/cabal2nix/blob/master/src/Cabal2Nix/PostProcess.hs#L44. I agree that this is a bit of a bug; I haven't come around to cleaning that up yet.